src/utils/getFilteredTodos.ts

```diff
@@ -0,0 +1,39 @@
+import { Todo } from '../types/Todo';
+import { Filters } from '../types/Filters';
+
+type Params = {
+  query: string;
+  selectedFilter: Filters;
+};
+
+export const getFilteredTodos = (
+  todos: Todo[],
+  { query, selectedFilter }: Params,
+) => {
+  let filteredTodos = [...todos];
+
+  filteredTodos = filteredTodos.filter(todo => {
+    switch (selectedFilter) {
+      case Filters.ACTIVE:
+        return !todo.completed;
+
+      case Filters.COMPLETED:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+
+  const normilizedQuery = query.toLowerCase().trim();
+
+  if (normilizedQuery) {
+    filteredTodos = filteredTodos.filter(({ title }) => {
+      const normilizedTitle = title.toLowerCase();
+
+      return normilizedTitle.includes(normilizedQuery);
+    });
+  }
```

Comment:
The search query filtering is applied after the status filter. This means that if a user selects the 'ACTIVE' filter and then enters a search query, the search will only be applied to the active todos, not to all todos. If this is the intended behavior, it would be good to clarify it in the function's documentation. If not, you should apply the search query filter to the original todos array before applying the status filter.